### ISSUE TITLE
Service worker keeps showing 'trying to install' when using `base` under injectManifest strategy

### ISSUE THREAD

**Issue by cloydlau** [vite-pwa/vite-plugin-pwa#656]
# Minimal reproduction
https://github.com/cloydlau/reproduction.git
1. Run `cd vite-project`
2. Run `pnpm i`
3. Run `pnpm dev`
4. See Application - Service workers

# Keeps showing 'trying to install'
![image](https://github.com/vite-pwa/vite-plugin-pwa/assets/31238760/7bf42abe-82b1-4865-aa27-9c470566ac3b)

# Comparison
- ✓ remove `base` under generateSW strategy
- ✓ use `base` under generateSW strategy
- ✓ remove `base` or set `base` to `'/'` under injectManifest strategy
- ✗ use `base` under injectManifest strategy

**Comment by userquin** [comment]
Nuxt returning / page content instead dev sw, I'm checking what's happening...

**Comment by cloydlau** [comment]
Hello, you can check `./vite-project` in that reproduction repo for a non-nuxt project,
both `nuxt + @vite-pwa/nuxt` and `vite + vite-plugin-pwa` can reproduce this issue.

**Comment by userquin** [comment]
there is a bug in the vite-plugin-pwa, missing scope/context/baseurl in the dev sw mapping: this entry should include the scope https://github.com/vite-pwa/vite-plugin-pwa/blob/main/src/plugins/dev.ts#L25

**Comment by userquin** [comment]
when using default base it is working since the logic just remove the `/`, I'm going to change the logic to checking using the scope and removing the prefix (L536), should fix both issues:

![imagen](https://github.com/vite-pwa/vite-plugin-pwa/assets/6311119/e3272a15-37ac-4c25-b4f2-3f4373a5af01)

**Comment by userquin** [comment]
@cloydlau I'm going to update dependencies in nuxt pwa, in the mean time you can install the new `v0.18.1` version in your repro and use `pnpm.overrides` adding `vite-plugin-pwa` to check it:
```ts
"devDependencies": {
    "@vite-pwa/nuxt": "^0.4.0",
    "nuxt": "^3.10.1",
    "typescript": "^5.3.3",
    "vite": "^5.1.1",
    "vite-plugin-pwa": "^0.18.1",
    "vue-tsc": "^1.8.27",
    "workbox-core": "^7.0.0",
    "workbox-precaching": "^7.0.0"
  },
  "pnpm": {
    "overrides": {
      "vite-plugin-pwa": "^0.18.1"
    }
  }
```

**Comment by cloydlau** [comment]
Seems working, thanks!

**Comment by userquin** [comment]
@cloydlau released `@vite-pwa/nuxt v0.5.0`
